### PR TITLE
Fix for #2423

### LINKED ISSUE #2423: max_waiting in consumer create request causes "consumer deliver subject has wildcards 10079"
## Defect
When `"max_waiting": 512` is supplied on the consumer create request, the server responds with an api error
`consumer deliver subject has wildcards 10079`

#### Versions of `nats-server` and affected client libraries used:

nats-server: v2.3.5-beta

#### OS/Container environment:

Windows

#### Steps or code to reproduce the issue:

```
CONSUMER.DURABLE.CREATE.streamname.durname

{
	"stream_name": "streamname",
	"config": {
		"durable_name": "durname",
		"deliver_subject": "_INBOX.YZsveC3tro2QrGNpGscOCt",
		"deliver_policy": "all",
		"opt_start_seq": 0,
		"ack_policy": "none",
		"ack_wait": 30000000000,
		"max_ack_pending": 0,
		"filter_subject": "subject.A",
		"replay_policy": "instant",
		"rate_limit_bps": 0,
		"flow_control": false,
		"max_waiting": 512
	}
}
```

#### Expected result:

Consumer is created.

#### Actual result:

Api Error

```
INFO: [18284] 2021/08/09 21:51:21.943117 [TRC] 127.0.0.1:55436 - cid:5 - "v2.11.6:java" - <<- [PUB $JS.API.STREAM.CREATE.stream _INBOX.YZsveC3tro2QrGNpGscOCt.YZsveC3tro2QrGNpGscOEd 130]
...
INFO: [18284] 2021/08/09 21:51:21.943117 [TRC] 127.0.0.1:55436 - cid:5 - "v2.11.6:java" - <<- MSG_PAYLOAD: ["{\"name\":\"stream\",\"subjects\":[\"subject.*\"],\"retention\":\"limits\",\"storage\":\"memory\",\"num_replicas\":1,\"no_ack\":false,\"discard\":\"old\"}"]
...
INFO: [18284] 2021/08/09 21:51:21.973127 [TRC] 127.0.0.1:55436 - cid:5 - "v2.11.6:java" - <<- MSG_PAYLOAD: ["{\"stream_name\":\"streamname\",\"config\":{\"durable_name\":\"durname\",\"deliver_subject\":\"_INBOX.YZsveC3tro2QrGNpGscOKj\",\"deliver_policy\":\"all\",\"opt_start_seq\":0,\"ack_policy\":\"none\",\"ack_wait\":30000000000,\"max_ack_pending\":0,\"filter_subject\":\"subject.A\",\"replay_policy\":\"instant\",\"rate_limit_bps\":0,\"flow_control\":false,\"max_waiting\":512}}"]
...
\\\"error\\\":{\\\"code\\\":400,\\\"err_code\\\":10079,\\\"description\\\":\\\"consumer deliver subject has wildcards\\\"}}\"}"]
```

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

Resolves #2423 

/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_test.go [hunk 1]
```diff
@@ -12551,6 +12551,43 @@ func TestJetStreamConsumerPendingBugWithKV(t *testing.T) {
...
+	// we got the wrong err about deliver subject having a wildcard.
+	_, err = js.AddConsumer("TEST", &nats.ConsumerConfig{
+		Durable:        "nowcerr",
+		DeliverSubject: "X",
+		MaxWaiting:     100,
+	})
+	if err == nil {
+		t.Fatalf("Expected an error but got none")
+	}
+	if !strings.Contains(err.Error(), "push mode can not set max waiting") {
```
Comment (ripienaar) [inline server/jetstream_test.go:12586]:
```
if !IsNatsError(err, JSConsumerPushMaxWaitingErr) {....}
```

Comment (derekcollison) [reply server/jetstream_test.go:12586]: Where is that defined?

Comment (derekcollison) [reply server/jetstream_test.go:12586]: I see it referenced in comments but do not see where that utility is defined.

Comment (ripienaar) [reply server/jetstream_test.go:12586]: `jetstream_errors.go`

Comment (derekcollison) [reply server/jetstream_test.go:12586]: Gotcha, but that does not work on client returned errors.

Comment (ripienaar) [reply server/jetstream_test.go:12586]: true true, missed that this was a client returned error.  go client did not yet implement the ADR

**Pull request merged**